src/store/actions/UserProfile.ts

```diff
@@ -0,0 +1,20 @@
+import { ReduxAction } from "types/redux";
+
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+export enum UserProfileActionTypes {
+  UPDATE_USER_PROFILE_DATA = "UPDATE_USER_PROFILE_DATA",
+}
+
+export type UpdateUserProfileDataAction = ReduxAction<
+  UserProfileActionTypes.UPDATE_USER_PROFILE_DATA,
+  { userProfile?: WithId<User> }
+>;
+
+export const updateUserProfileData = (userProfile?: WithId<User>) => ({
+  type: UserProfileActionTypes.UPDATE_USER_PROFILE_DATA,
+  payload: { userProfile },
+});
```

Comment:
I believe this should be something like this?

```suggestion
export const updateUserProfileData: UpdateUserProfileDataAction = (userProfile) => ({
  type: UserProfileActionTypes.UPDATE_USER_PROFILE_DATA,
  payload: { userProfile },
});
```